server/service/src/print/label.rs

```diff
@@ -43,6 +46,66 @@ pub fn print_qr_code(
     printer.send_string(payload, Mode::Print)
 }
 
+#[derive(serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PrescriptionLabelData {
+    item_details: String, // usually the amount of units + the item name e.g. "10Tabs Paracetamol 500mg Tablets"
+    item_directions: String,
+    patient_details: String, // e.g patient name, possibly code etc.
+    warning: Option<String>, // Some items come with a defined warning (OG field "Message") that should be printed on all labels regardless of directions e.g. avoid sun exposure, avoid alcohol...
+    details: String, // General details to include e.g. store name, prescriber name, date/time...
+}
+
+pub fn print_prescription_label(
+    settings: LabelPrinterSettingNode,
+    label_data: PrescriptionLabelData,
+) -> Result<String> {
+    let PrescriptionLabelData {
+        item_details,
+        item_directions,
+        patient_details,
+        warning,
+        details,
+    } = label_data;
+    let warning = warning.unwrap_or_default();
+
+    let payload = format!(
+        r#"
+^XA
+^FX CI command parameters:
+^FX - encoding (28 = UTF-8)
+^CI28
+
+^A0,25
+^FO20,10
+^FB550,1,0,C
+^FD{item_details}\&^FS
+
+^FO20,50
+^GB550,2,2^FS
+
+^A0,25
+^FO20,60
+^FB550,6,0,L
+^FD{item_directions}\&{warning}^FS
+
+^FO20,210
+^GB550,2,2^FS
+
+^A0,25
+^FO20,220
+^FD{patient_details}^FS
+
+^A0,25
+^FO20,250
+^FD{details}^FS
+^XZ
+"#
+    );
+    let printer = Jetdirect::new(settings.address, settings.port);
+    printer.send_string(payload, Mode::Print)
```

Comment:
Just running the same way as the qr codes, however you can literally just make a regular http POST with the ZPL text to print labels. At least for zebra printers anyway.